src/Hosting/TestHost/src/ClientHandler.cs

```diff
@@ -65,7 +66,19 @@ internal ClientHandler(PathString pathBase, ApplicationWrapper application)
             var contextBuilder = new HttpContextBuilder(_application, AllowSynchronousIO, PreserveExecutionContext);
 
             var requestContent = request.Content ?? new StreamContent(Stream.Null);
-            var body = await requestContent.ReadAsStreamAsync();
+
+            // This is odd but required for backwards compat.
+            // If StreamContent is passed in then seek to beginning.
+            if (requestContent is StreamContent)
+            {
+                var body = await requestContent.ReadAsStreamAsync();
```

Comment:
What's weird about this logic is that it assumes the body will be re-used again when you do StreamContent.CopyToAsync.

Reply:
Not having it will break anyone who relied on https://github.com/aspnet/AspNetCore/pull/15591/files#diff-5e75529c0ee5684e3ab7c04e5a7f63a9L121

Reply:
To make it a bit more solid it makes sense to use the Stream instead of rewinding it and throwing it away. Something like:

```C#
var requestPipe = new Pipe();

_ = Task.Run(async () =>
{
    if (requestContent is StreamContent)
    {
        var body = await requestContent.ReadAsStreamAsync();
        if (body.CanSeek)
        {
            // This body may have been consumed before, rewind it.
            body.Seek(0, SeekOrigin.Begin);
        }
        
        await body.CopyToAsync(requestPipe.Writer); // This works 😄 
    }
    else
    {
        await requestContent.CopyToAsync(requestPipe.Writer.AsStream());
    }
    
    await requestPipe.Writer.CompleteAsync();
});

req.Body = new AsyncStreamWrapper(requestPipe.Reader.AsStream(), () => contextBuilder.AllowSynchronousIO);
```